lib/src/upZenohClient.cpp

```diff
@@ -42,6 +64,8 @@ std::shared_ptr<UpZenohClient> UpZenohClient::instance(void) noexcept {
         }
 
         handle = std::make_shared<UpZenohClient>(ConstructToken());
+        auto rpc_handle = static_pointer_cast<ZenohRpcClient>(handle);
+        rpc_handle->clientAuthority = src_authority;
```

Comment:
If taking a std::optional for the source, then it would be checked here:
```suggestion
        if (!src_authority) {
            return nullptr;
        }
        auto rpc_handle = static_pointer_cast<ZenohRpcClient>(handle);
        rpc_handle->clientAuthority = *src_authority;
```

Reply:
I intended to make the src_authority optional. What is is the point of returning nullptr if the optional is not provided? Seems better to just not make it optional. Or, am I misunderstanding your point?

Reply:
I think you might be missing the point. The source parameter is only required when constructing a new instance. When a valid instance already exists, it is optional (or really it's an error to provide a source when the instance is already initialized). Returning nullptr is the same error propagation strategy as when one of the lower layers fails to initialize (see a couple lines down in the file).

The goal here is to have one implementation of the instance function.